readthedocs/proxito/views/serve.py

```diff
@@ -178,22 +180,18 @@ def get(self, request, proxito_path, template_name='404.html'):
             filename=kwargs.get('filename', ''),
         )
 
-        # Always add a `/` to the filename to match our old logic:
+        # ``filepath`` is the path without ``/<lang>/<version>`` and without
+        # query, starting with a ``/``. This matches our old logic:
         # https://github.com/readthedocs/readthedocs.org/blob/4b09c7a0ab45cd894c3373f7f07bad7161e4b223/readthedocs/redirects/utils.py#L60
-        redirect_filename = filename
-        if lang_slug and version_slug:
-            redirect_filename = '/' + filename
+        # We parse ``filename`` to remove the query from it
+        schema, netloc, path, params, query, fragments = urlparse(filename)
+        filepath = path
 
-        # Check and perform redirects on 404 handler
-        redirect_path, http_status = self.get_redirect(
-            project=final_project,
-            lang_slug=lang_slug,
-            version_slug=version_slug,
-            filename=redirect_filename,
-            full_path=proxito_path,
-        )
-        if redirect_path and http_status:
-            return self.get_redirect_response(request, redirect_path, http_status)
+        # we can't check for lang and version here to decide if we need to add
+        # the ``/`` or not because ``/install.html`` is a valid path to use as
+        # redirect and does not include lang and version on it. It should be
+        # fine always adding the ``/`` to the beginning.
+        filepath = '/' + filepath.lstrip('/')
```

Comment:
I added a small protection for `filepath` that already starts with `/` here. This is the case of crossdomain redirects (that I couldn't make them work yet, BTW)

Reply:
Wouldn't we want this protection before we parse it with urlparse?

Reply:
I'm not sure I have a good answer for that. What would be the case that we are covering?